ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/visitor/infraestructure/antlr/visitors/JavaMethodVisitor.java

```diff
@@ -0,0 +1,155 @@
+import org.antlr.v4.runtime.ParserRuleContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class JavaMethodVisitor extends JavaParserBaseVisitor<MethodInfo>
+        implements IMethodAnalyzerVisitor<ParserRuleContext> {
+
+    private final CodeMetricsAnalyzer codeAnalyzer;
+    private final ComplexityMetricsAnalyzer complexityAnalyzer;
+    private final DependencyAnalyzer dependencyAnalyzer;
+
+    public JavaMethodVisitor() {
+        this.codeAnalyzer = new CodeMetricsAnalyzer();
+        this.complexityAnalyzer = new ComplexityMetricsAnalyzer();
+        this.dependencyAnalyzer = new DependencyAnalyzer();
+    }
+
+    public JavaMethodVisitor(
+            CodeMetricsAnalyzer codeAnalyzer,
+            ComplexityMetricsAnalyzer complexityAnalyzer,
+            DependencyAnalyzer dependencyAnalyzer
+    ) {
+        this.codeAnalyzer = codeAnalyzer;
+        this.complexityAnalyzer = complexityAnalyzer;
+        this.dependencyAnalyzer = dependencyAnalyzer;
+    }
+
+    @Override
+    public MethodInfo visitMethod(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx -> {
+            String name = methodCtx.identifier().getText();
+            String returnType = getReturnType(methodCtx);
+            List<String> modifiers = getMethodModifiers(methodCtx);
+            List<ParameterInfo> parameters = getParameters(methodCtx);
+            ComplexityMetrics complexityMetrics = getComplexityMetrics(methodCtx);
+            CodeMetrics codeMetrics = getCodeMetrics(methodCtx);
+            DependencyInfo dependencies = getDependencyInfo(methodCtx);
+
+            return new MethodInfo(
+                    name,
+                    returnType,
+                    modifiers,
+                    parameters,
+                    complexityMetrics,
+                    codeMetrics,
+                    dependencies
+            );
+        }, null);
+    }
+
+    @Override
+    public CodeMetrics getCodeMetrics(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx ->
+                        codeAnalyzer.analyze(methodCtx),
+                new CodeMetrics(0, 0, 0, 0)
+        );
+    }
+
+    @Override
+    public ComplexityMetrics getComplexityMetrics(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx ->
+                        complexityAnalyzer.analyze(methodCtx),
+                new ComplexityMetrics(1, 0, 0)
+        );
+    }
+
+    @Override
+    public DependencyInfo getDependencyInfo(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx ->
+                        dependencyAnalyzer.analyze(methodCtx),
+                new DependencyInfo(new ArrayList<>(), new ArrayList<>(), new ArrayList<>())
+        );
+    }
+
+    @Override
+    public List<String> getMethodModifiers(ParserRuleContext ctx) {
```

Comment:
To have a clean and understandable code, it would be good to divide this method since it is recommended to have between 20 to 25 lines of code. You could divide the responsibility into different methods and have a Map variable to avoid having many else-ifs.